thingif/src/main/java/com/kii/thingif/command/Action.java

```diff
@@ -30,5 +24,12 @@
  *
  */
 public interface Action {
+    /**
```

Comment:
> So as a result, we left 2 ways to developer. I think less choices is simple.

As you said, less choices are simple and better. I agree with it.

However, validating fields problem is different from this. As I said in https://github.com/KiiPlatform/thing-if-AndroidSDK/pull/293#discussion_r108847958, A class should validate the field the class has. This is simple way to use an instance of the class.

I already agree with your decision to adopt interface at https://github.com/KiiPlatform/thing-if-AndroidSDK/pull/293#discussion_r108845207 .

In this case, adopting interface give more benefits to application developers than validating fields in a class so I agree with you opinion.